### PR TITLE
Makes red dot sight reduce aim mode fire delay by 50%

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request
Makes red dot sight reduce aim mode fire rate debuff by 50%.

## Why It's Good For The Game
MJP said it's good, ask them if you have questions
Contender to magnetic harness for the rail attachment slot

## Changelog
:cl:
balance: red dot reduces fire delay from aim mode by 50%
/:cl:

### CODE REVIEW

**Comment by KornFlaks** [comment]
RDS or Mag harn.... actually not a bad idea. If this gets less marines to use mag harn so I can strong acid more guns, that would be great.

Although 50% is a bit much. 20% or 25% might be better.

**Comment by EvilMelonTheSecond** [comment]
Finally a reason to use anything other than mag harn

**Comment by Losenis** [comment]
Based, now we need to give the miniscope the ability to move while zoomed.

**Comment by SplinterGP** [comment]
> 
> 
> Based, now we need to give the miniscope the ability to move while zoomed.

inbound

**Comment by Losenis** [comment]
> inbound

image

**Comment by Walarks** [comment]
> RDS or Mag harn.... actually not a bad idea. If this gets less marines to use mag harn so I can strong acid more guns, that would be great.
> 
> Although 50% is a bit much. 20% or 25% might be better.

25% might be too small of a value to make a difference compared to mag harness. If it proves to be too good, I can reduce it to 40%

**Comment by Losenis** [comment]
Stick to 50%, it keeps the values within 0.05 multiples for the sake of firerates for values like 0.1 aim delay.

File: code/modules/projectiles/gun_attachables.dm [hunk 1]
```diff
@@ -582,13 +582,24 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
...
 
+/obj/item/attachable/reddot/attach_to_gun(obj/item/weapon/gun/gun_to_attach, mob/user)
+	. = ..()
+	cached_fire_rate = master_gun.aim_fire_delay * aim_mode_fire_rate_debuff_reduction
+	master_gun.aim_fire_delay -= cached_fire_rate
+
+/obj/item/attachable/reddot/detach_from_master_gun(mob/user)
+	master_gun.aim_fire_delay += cached_fire_rate
+	cached_fire_rate = 0
+	return ..()
```
Comment (SplinterGP) [inline code/modules/projectiles/gun_attachables.dm:602]: bad code, add it to attachment code, and why do you need this cached

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_attachables.dm (hunk 1 above)
Comment (Walarks) [reply code/modules/projectiles/gun_attachables.dm]: changed

File: code/modules/projectiles/gun_attachables.dm [hunk 2]
```diff
@@ -250,6 +260,8 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	master_gun.scatter_unwielded			-= scatter_unwielded_mod
 	master_gun.aim_speed_modifier			-= initial(master_gun.aim_speed_modifier)*aim_mode_movement_mult
 	master_gun.iff_marine_damage_falloff	-= shot_marine_damage_falloff
+	master_gun.aim_fire_delay 				+= cached_aim_mode_debuff_fire_rate
```
Comment (QualityVan) [inline code/modules/projectiles/gun_attachables.dm:263]:
```suggestion
	master_gun.aim_fire_delay 				+= cached_aim_mode_debuff_fire_rate
	if(CHECK_BITFIELD(master_gun.flags_gun_features, GUN_IS_AIMING))
		master_gun.modify_fire_delay(cached_aim_mode_debuff_fire_rate)
```
This'll catch the current issue where you can reduce a gun's fire delay to zero by removing the RDS while aiming.

**Comment by TiviPlus** [comment]
See #7710

**Pull request merged**